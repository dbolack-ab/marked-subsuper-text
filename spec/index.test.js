import { marked } from 'marked';
import subSuper from '../src/index.js';

function trimLines(s) {
  return s.split('\n').map(l => l.trim()).join('\n');
}

describe('subsuper-text', () => {
  beforeEach(() => {
    marked.setOptions(marked.getDefaults());
  });

  test('Superscript', () => {
    marked.use({ extensions: [ subSuper ] });
    expect(marked(trimLines(`This is ^superscript^.`))).toMatchSnapshot();
  });

  test('Subscript', () => {
    marked.use({ extensions: [ subSuper ] });
    expect(marked(trimLines(`This is ^^subscript^^.`))).toMatchSnapshot();
  });

  test('Sub and Super', () => {
    marked.use({ extensions: [ subSuper ] });
    expect(marked(trimLines(`This is ^^sub^^ and this is ^super^.`))).toMatchSnapshot();
  });

});
