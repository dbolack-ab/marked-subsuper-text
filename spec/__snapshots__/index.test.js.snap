// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`subsuper-text Sub and Super 1`] = `
"<p>This is ^^sub^^ and this is ^super^.</p>
"
`;

exports[`subsuper-text Subscript 1`] = `
"<p>This is ^^subscript^^.</p>
"
`;

exports[`subsuper-text Superscript 1`] = `
"<p>This is ^superscript^.</p>
"
`;
